### PR TITLE
Mintegral - Initial adapters release - 5.8.4.0.0

### PR DESCRIPTION
**Pull request by Mintegral-official**

### CODE REVIEW

File: Mintegral/MintegralBannerCustomEvent.m [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    
+    if (errorMsg) {
+        
+        NSError *error = [NSError errorWithDomain:kMintegralErrorDomain code:MintegralErrorBannerParaUnresolveable userInfo:@{NSLocalizedDescriptionKey : errorMsg}];
+        if ([self.description respondsToSelector:@selector(bannerCustomEvent: didFailToLoadAdWithError:)]) {
+            MPLogAdEvent([MPLogEvent adLoadFailedForAdapter:NSStringFromClass(self.class) error:error], nil);
+            [self.delegate bannerCustomEvent:self didFailToLoadAdWithError:error];
+        }
+        return;
+    }
```
[the remark this answers is not in the document]
Comment (Mintegral-official) [reply Mintegral/MintegralBannerCustomEvent.m]: will do

File: Mintegral/MintegralAdapterConfiguration.m [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
++(void)sdkInitialized{
+#ifdef DEBUG
+    if (DEBUG) {
+        MPLogInfo(@"The version of current Mintegral Adapter is: %@",MintegralAdapterVersion);
+    }
+#endif
+    Class class = NSClassFromString(@"MTGSDK");
+    SEL selector = NSSelectorFromString(@"setChannelFlag:");
+    NSString *pluginNumber = @"Y+H6DFttYrPQYcIA+F2F+F5/Hv==";
```
[the remark this answers is not in the document]
Comment (Mintegral-official) [reply Mintegral/MintegralAdapterConfiguration.m]: pluginNumber will never change. will do.

File: Mintegral/MintegralAdapterConfiguration.m [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
++(BOOL)isSDKInitialized{
+    return mintegralSDKInitialized;
+}
+
++(void)sdkInitialized{
+#ifdef DEBUG
+    if (DEBUG) {
+        MPLogInfo(@"The version of current Mintegral Adapter is: %@",MintegralAdapterVersion);
+    }
+#endif
```
[the remark this answers is not in the document]
Comment (Mintegral-official) [reply Mintegral/MintegralAdapterConfiguration.m]: I will delete in the latest commit

File: Mintegral/MintegralAdapterConfiguration.m [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+- (NSString *)adapterVersion {
+    return MintegralAdapterVersion;
+}
+
+- (NSString *)biddingToken {
+    return [MTGBiddingSDK buyerUID];
+}
+
+- (NSString *)moPubNetworkName {
+    return @"Mintegral";
```
[the remark this answers is not in the document]
Comment (Mintegral-official) [reply Mintegral/MintegralAdapterConfiguration.m]: will do

File: Mintegral/MintegralAdapterConfiguration.m [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
++(void)setGDPRInfo:(NSDictionary *)info{
+    if([[MoPub sharedInstance] canCollectPersonalInfo])
+    {
+        [[MTGSDK sharedInstance] setConsentStatus:YES];
+        NSString *privateInfo = @"Can send GDPR";
+        MPLogInfo(@"%@", privateInfo);
+    }else{
+        [[MTGSDK sharedInstance] setConsentStatus:NO];
+        NSString *privateInfo = @"Cannot send GDPR";
+        MPLogInfo(@"%@", privateInfo);
```
[the remark this answers is not in the document]
Comment (Mintegral-official) [reply Mintegral/MintegralAdapterConfiguration.m]: will do

File: Mintegral/MintegralBannerCustomEvent.m [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    _bannerAdView = [[MTGBannerAdView alloc] initBannerAdViewWithAdSize:size unitId:unitId rootViewController:vc];
+    _bannerAdView.delegate = self;
+    
+    self.adm = adMarkup;
+    if (self.adm) {
+        MPLogInfo(@"Loading Mintegral Banner ad markup for Advanced Bidding");
+        [_bannerAdView loadBannerAdWithBidToken:self.adm];
+        MPLogAdEvent([MPLogEvent adLoadAttemptForAdapter:NSStringFromClass(self.class) dspCreativeId:nil dspName:nil], self.adUnitId);
+    }else{
+        MPLogInfo(@"Loading Mintegral native ad");
```
[the remark this answers is not in the document]
Comment (Mintegral-official) [reply Mintegral/MintegralBannerCustomEvent.m]: will do

File: Mintegral/MintegralNativeAdRenderer.m [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+#import <MTGSDK/MTGAdChoicesView.h>
+@interface MintegralNativeAdRenderer () <MPNativeAdRendererSettings>
+
+@property (nonatomic, strong) UIView<MPNativeAdRendering> *adView;
+@property (nonatomic, strong) MintegralNativeAdAdapter *adapter;
+@property (nonatomic, strong) Class renderingViewClass;
+
+@end
+
+@implementation MintegralNativeAdRenderer
```
[the remark this answers is not in the document]
Comment (Mintegral-official) [reply Mintegral/MintegralNativeAdRenderer.m]: will do

File: Mintegral/MintegralBannerCustomEvent.m [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        NSError *error = [NSError errorWithDomain:kMintegralErrorDomain code:MintegralErrorBannerParaUnresolveable userInfo:@{NSLocalizedDescriptionKey : errorMsg}];
+        if ([self.description respondsToSelector:@selector(bannerCustomEvent: didFailToLoadAdWithError:)]) {
+            MPLogAdEvent([MPLogEvent adLoadFailedForAdapter:NSStringFromClass(self.class) error:error], nil);
+            [self.delegate bannerCustomEvent:self didFailToLoadAdWithError:error];
+        }
+        return;
+    }
+    [MintegralAdapterConfiguration initializeMintegral:info setAppID:appId appKey:appKey];
+    _adUnitId = unitId;
+    _currentSize = size;
```
[the remark this answers is not in the document]
Comment (Mintegral-official) [reply Mintegral/MintegralBannerCustomEvent.m]: We can accept any banner size sent from mopub and server suitable ads.

**Comment by Mintegral-official** [comment]
Thanks for your observant and patient review much.

**Comment by chauduyphanvu** [comment]
Released in https://github.com/mopub/mopub-ios-mediation/commit/870cd3c4980080b9e00671dbe32b130f9131154f.

**Pull request closed without merging**